src/features/forecastapi/forecast-api.ts

```diff
@@ -0,0 +1,32 @@
+import { TimeRegistrations } from "@functions/schema";
+import fetch, { Headers } from "node-fetch";
+
+/**
+ *  Namespace for forecast-api
+ */
+namespace ForecastApiUtilities {
+  const header ={
+    "X-FORECAST-API-KEY": `${process.env.X_FORECAST_API_KEY}`
+  };
+
+  const headers = new Headers(header);
+
+  /**
+   * Get all allocations after yesterday
+   * 
+   * @param yesterday yesterday
+   * @returns all allocations after yesterday
+   */
+  export const getTimeRegistrations = async (dayBeforeYesterday: string): Promise<TimeRegistrations[]> => {
+    try {
+      const dayBeforeYesterdayUrl = dayBeforeYesterday.replace(/[-]/g, "");
+      const result = await fetch(`${process.env.forecast_v3_url}time_registrations?date_after=${dayBeforeYesterdayUrl}`, { headers: headers });
+      const timeRegistrations: TimeRegistrations[] = await result.json();
+      return timeRegistrations.filter(timeRegistrations => timeRegistrations.non_project_time !== null);
+    } catch(error) {
+      console.error("Error while loading time registrations");
+      Promise.reject(error);
+    }
+  };
```

Comment:
Add another call for fetching all non_project_time types from forecast (https://api.forecast.it/api/v1/non_project_time). This will give you a list of all configured non-project time types we have. Each entry contains and ID (for example 228255 is vacation ID) that can be used to filter the entries. Entries from that endpoint all contain variable called "is_internal_time". If this is false then we know that all those entries are the ones we should use for filtering time entries.

And since some time allocation should be marked to internal times (like misc and management related stuff), which will be marked as non_project_time, it is better to use unique ID's to verify should the user get messages or not.

Reply:
Done